File: LeetCode/Python/Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        ss = []
        for c in s:
            if c == '(' or c == '{' or c == '[':
                ss.append(c)
            else:
                if len(ss) == 0:
                    return False
                else:
                    if c == ')':
                        if ss[len(ss)-1] == '(':
                            ss.pop()
                        else:
                            return False
                    elif c == '}':
                        if ss[len(ss)-1] == '{':
                            ss.pop()
                        else:
                            return False
                    else:
                        if ss[len(ss)-1] == '[':
                            ss.pop()
                        else:
                            return False
        if len(ss) != 0:
            return False
        return True

File: LeetCode/C++/Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> ss;
        for(char c:s){
            if(c == '(' || c == '{' || c == '['){
                ss.push(c);
            }
            else{
                if(c == ')'){
                    if(ss.empty()){
                        return false;
                    }
                    else{
                        if(ss.top() == '('){
                            ss.pop();
                        }
                        else{
                            return false;
                        }
                    }
                }
                else if(c == '}'){
                    if(ss.empty()){
                        return false;
                    }
                    else{
                        if(ss.top() == '{'){
                            ss.pop();
                        }
                        else{
                            return false;
                        }
                    }
                }
                else{
                    if(ss.empty()){
                        return false;
                    }
                    else{
                        if(ss.top() == '['){
                            ss.pop();
                        }
                        else{
                            return false;
                        }
                    }
                }
            }
        }
        if(!ss.empty())return false;
        return true;
    }
};